src/util/fs_utils.js

```diff
@@ -219,31 +218,34 @@ async function file_delete(file_name) {
     }
 }
 
-function full_dir_copy(src, dst, filter_regex) {
-    return P.fromCallback(callback => {
-        ncp.limit = 10;
-        const ncp_options = {};
-        if (filter_regex) {
-            //this regexp will filter out files that matches, except path.
-            const ncp_filter_regex = new RegExp(filter_regex);
-            const ncp_filter_function = input => {
-                if (input.indexOf('/') > 0) {
-                    return false;
-                } else if (ncp_filter_regex.test(input)) {
-                    return false;
-                } else {
-                    return true;
-                }
-            };
-            ncp_options.filter = ncp_filter_function;
+async function _filtered_file_copy(src, dst, filter_regex) {
+    await fsp.mkdir(dst, { recursive: true });
+    const files = await fsp.readdir(src, { withFileTypes: true });
+
+    for (const file of files) {
+        const src_path = path.join(src, file.name);
+        const dst_path = path.join(dst, file.name);
+
+        if (filter_regex && filter_regex.test(file.name)) {
+            continue;
         }
-        if (!src || !dst) {
-            throw new Error('Both src and dst must be given');
+
+        if (file.isDirectory()) {
+            await _filtered_file_copy(src_path, dst_path);
+        } else if (file.isFile()) {
+            await fsp.copyFile(src_path, dst_path);
         }
-        ncp(src, dst, ncp_options, callback);
-    }).then(() => {
-        // do nothing. 
-    });
+    }
+}
+
+async function full_dir_copy(src, dst, filter_regex) {
```

Comment:
Also, type script for the variables will be nice.
/**
     * full_dir_copy - bla bla bla.
     * 
     * @param {string} src
     * @param {string} dst
     * @param {filter_regex} string
     * @returns {Promise<void>
     */